e2e/shannon_relay_test.go

```diff
@@ -4,102 +4,97 @@ package e2e
 
 import (
 	"bytes"
-	"context"
 	"encoding/json"
 	"errors"
 	"fmt"
 	"io"
-	"math/rand"
 	"net/http"
-	"net/url"
+	"os"
 	"testing"
-	"time"
 
-	"github.com/pokt-network/poktroll/pkg/polylog"
-	"github.com/pokt-network/poktroll/pkg/polylog/polyzero"
 	"github.com/stretchr/testify/require"
-
-	"github.com/buildwithgrove/path/config"
-	"github.com/buildwithgrove/path/relayer"
-	"github.com/buildwithgrove/path/relayer/shannon"
-	"github.com/buildwithgrove/path/request"
 )
 
-const configPath = ".config.test.yaml"
+// localdev.me is a hosted domain that resolves to 127.0.0.1 (localhost).
+// This allows a subdomain to be specified without modifying /etc/hosts.
+// It is hosted by AWS. See https://codeengineered.com/blog/2022/localdev-me/
+const localdevMe = "localdev.me"
 
-var testTimeout = 20 * time.Second
+// When the ephemeral PATH Docker container is running it exposes a dynamically
+// assigned port. This global variable is used to capture the port number.
+var pathPort string
 
-// TODO_INCOMPLETE: add an action to the CI for running E2E tests, which at the minimum
-// includes using the e2e tag and defining secrets to be used as environment variables, e.g. gateway/signer private key and address.
-//
-// TODO_IMPROVE: use gocuke (github.com/regen-network/gocuke) for defining and running E2E tests.
-func TestShannonRelay(t *testing.T) {
+func TestMain(m *testing.M) {
+	// Initialize the ephemeral PATH Docker container
+	pool, resource, containerPort := setupPathDocker()
+
+	// Assign the port the container is listening on to the global variable
+	pathPort = containerPort
+
+	// Run PATH E2E Shannon relay tests
+	exitCode := m.Run()
+
+	// Cleanup the ephemeral PATH Docker container
+	cleanupPathDocker(m, pool, resource)
+
+	// Exit with the test result
+	os.Exit(exitCode)
+}
+
+func Test_ShannonRelay(t *testing.T) {
 	tests := []struct {
-		name      string
-		serviceID relayer.ServiceID
-		relayID   string
-		httpReq   *http.Request
+		name         string
+		reqMethod    string
+		reqPath      string
+		serviceID    string
+		serviceAlias string
+		relayID      string
+		body         string
```

Comment:
```suggestion

		expectedBodyResponse         string
```

Reply:
That's not what this variable is; it's the body that is sent as part of the request. The test doesn't care about the response currently; it only cares that at least one request succeeds.

Reply:
#PUC